### PR TITLE
[RPA-944] Fixes issue where negative number of rows/columns was accepted

### PR DESCRIPTION
**Pull request by bhszewczyk**
Number of columns/rows to insert provided in the input cannot be negative - wrong input error message is thrown.

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts [hunk 1]
```diff
@@ -618,4 +628,8 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         }
         throw new Error(ExcelErrorMessage.tableNotFoundIncorrectInput());
     }
+
+    private isPositiveInteger(number: number): boolean {
```
Comment (MRSIMM0) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: In every occurrence of this function you negate the result. So maybe isNegativeInteager function would be better?

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts [hunk 2]
```diff
@@ -218,10 +219,11 @@ export default class ExcelActionHandler extends StatefulActionHandler {
     async insertColumnsAfter(input: ExcelInsertColumnsActionInput): Promise<void> {
         if (input.worksheet) this.checkIsWorksheetNameCorrect(input.worksheet, true);
         const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
-
+         
         try {
             const column = this.getColumnCoordinate(input.column);
             const amount = input.amount;
+            if (!this.isPositiveInteger(amount)) throw new Error();
```
Comment (preludium) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: provide error message plx

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts [hunk 3]
```diff
@@ -618,4 +628,8 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         }
         throw new Error(ExcelErrorMessage.tableNotFoundIncorrectInput());
     }
+
+    private isPositiveInteger(number: number): boolean {
+        return (number > 0 && Number.isInteger(number))
```
Comment (preludium) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: isInteger check should occur before checking value

**Review by preludium — APPROVED** [review]

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts (hunk 2 above)
Comment (bhszewczyk) [reply runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: It is not being provided not to double the error message thrown in a catch. :)

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts (hunk 1 above)
Comment (bhszewczyk) [reply runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: Ok, changed it. The only thing is that 0 is neither negative or positive number so isNegative now check if the number is below 0, so 0 is acceptable (previously it wasn't), action will be executed, but nothing will happen. Checking if is positive allowed me to exclude 0 as well.

**Pull request merged**